binary-search-trees/01-Binary-Search-Trees.md

```diff
@@ -1009,13 +829,49 @@ def delete_helper(self, current_root, key):
     return current_root
 
 def delete(self, key):
+    # if the tree is empty
     if not self.root:
+        # exit the function
         return
-    self.root = self.delete_helper(self.root, key)
 
+    # call our recursive helper on the root
+    self.root = self.delete_helper(self.root, key)
 ```
+</summary>
 
-##### !end-explanation 
+## Reflection
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: paragraph
+* id: b4148882-340f-40e2-87c0-0b2b1265fc49
+* title: Binary Search Trees Reflection
+* points: 1
```

Comment:
I'm not sure if this has any affect on grading, but when I've been looking through Learn for checking people's graduation requirement completion for CS Fun, some lessons appear to be "incomplete" because they haven't filled out the reflection question. Would it be best to change the points to 0 for the reflection-style questions to avoid any potential confusion for instructors in the future?

Reply:
Ah yes, I actually put this as 1 because I wanted students to complete it, but I can change it to 0